swarm/swarm.go

```diff
@@ -179,6 +181,7 @@ func NewSwarm(config *api.Config, mockStore *mock.NodeStore) (self *Swarm, err e
 			return nil, err
 		}
 		self.swap = swap.New(balancesStore)
+		self.accountingMetrics = protocols.SetupAccountingMetrics(10*time.Second, filepath.Join(config.Path, "metrics.db"))
```

Comment:
should this not be managed within swap?

Reply:
no, because this mostly measures things which are happening inside the `p2p/protocols` package.  Swap maintains balances. These metrics contains:
* A **cumulative** metric of how many units have been credited/debited
* The amount of bytes received/sent
* The amount of messages (unitary)
* The amount of peer drops due to limit exceeds

Reply:
Some of these things are only available here (like amount of bytes)